internal/controller/kubecostextractor_controller.go

```diff
@@ -0,0 +1,380 @@
+	}
+
+	r.RunOnInterval(ctx, req.NamespacedName.String(), kubecost.Spec.GetInterval(), func(ctx context.Context) (done bool, err error) {
+		// Always patch object when exiting this function, so we can persist any object changes.
+		defer func() {
+			if err := scope.PatchObject(); err != nil && reterr == nil {
+				reterr = err
+			}
+		}()
+		recommendations, err := r.getRecommendationAttributes(ctx, kubecostService, kubecost.Spec.GetPort(), kubecost.Spec.GetInterval(), recommendationThreshold)
+		if err != nil {
+			logger.Error(err, "Unable to fetch recommendations")
+			utils.MarkCondition(kubecost.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionFalse, v1alpha1.ErrorConditionReason, err.Error())
+			return false, nil
+		}
+		clusterCostAttr, err := r.getClusterCost(ctx, kubecostService, kubecost.Spec.GetPort(), kubecost.Spec.GetInterval())
+		if err != nil {
+			logger.Error(err, "Unable to fetch cluster cost")
+			utils.MarkCondition(kubecost.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionFalse, v1alpha1.ErrorConditionReason, err.Error())
+			return false, nil
+		}
+		namespacesCostAtrr, err := r.getNamespacesCost(ctx, kubecostService, kubecost.Spec.GetPort(), kubecost.Spec.GetInterval())
+		if err != nil {
+			logger.Error(err, "Unable to fetch namespacesCostAtrr cost")
+			utils.MarkCondition(kubecost.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionFalse, v1alpha1.ErrorConditionReason, err.Error())
+			return false, nil
+		}
+
+		// nothing for specified time window
+		if clusterCostAttr == nil && namespacesCostAtrr == nil && recommendations == nil {
+			utils.MarkCondition(kubecost.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionTrue, v1alpha1.ReadyConditionReason, "")
+			return false, nil
+		}
+
+		if _, err := r.ExtConsoleClient.IngestClusterCost(console.CostIngestAttributes{
+			Cluster:         clusterCostAttr,
+			Namespaces:      namespacesCostAtrr,
+			Recommendations: recommendations,
+		}); err != nil {
+			logger.Error(err, "Unable to ingest cluster cost")
+			utils.MarkCondition(kubecost.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionFalse, v1alpha1.ErrorConditionReason, err.Error())
+			return false, nil
+		}
+		utils.MarkCondition(kubecost.SetCondition, v1alpha1.ReadyConditionType, v1.ConditionTrue, v1alpha1.ReadyConditionReason, "")
+		return false, nil
+	})
+
+	return ctrl.Result{}, nil
+}
+
+func (r *KubecostExtractorReconciler) getAllocation(ctx context.Context, srv *corev1.Service, servicePort, aggregate string) (*allocationResponse, error) {
+	now := time.Now()
+	oneMonthBefore := now.AddDate(0, -1, 0) // 0 years, -1 month, 0 days
+	window := fmt.Sprintf("%d,%d", oneMonthBefore.Unix(), now.Unix())
+
+	queryParams := map[string]string{
+		"window":     window,
+		"aggregate":  aggregate,
+		"accumulate": "true",
+	}
+
+	bytes, err := r.KubeClient.CoreV1().Services(srv.Namespace).ProxyGet("", srv.Name, servicePort, "/model/allocation", queryParams).DoRaw(ctx)
```

Comment:
Why do we need to use a proxy get here?  Shouldn't we just be able to use the service dns itself?

Reply:
Generally, I got this from the `kubectl-cost` https://github.com/kubecost/kubectl-cost/blob/main/pkg/query/allocation.go#L38

Finally `DoRaw` makes the regular HTTP call